core/grant/service.go

```diff
@@ -533,3 +663,28 @@ func reduceGrantsByProviderRole(rc domain.ResourceConfig, grants []*domain.Grant
 
 	return
 }
+
+func getGrantIDs(grants []domain.Grant) []string {
+	var ids []string
+	for _, g := range grants {
+		ids = append(ids, g.ID)
+	}
+	return ids
+}
+
+func structToMap(item interface{}) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+
+	if item != nil {
+		jsonString, err := json.Marshal(item)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal(jsonString, &result); err != nil {
+			return nil, err
+		}
+	}
+
+	return result, nil
+}
```

Comment:
this same functionality is also here - https://github.com/goto/guardian/blob/main/domain/appeal.go#L264. Can we keep this in utils and reuse it.